geode-serialization/src/test/java/org/apache/geode/internal/serialization/filter/ReflectiveFacadeGlobalSerialFilterFactoryTest.java

```diff
@@ -41,8 +49,9 @@ public void constructsDelegatingGlobalSerialFilter() {
   public void delegatesToObjectInputFilterApiToCreateObjectInputFilter()
       throws InvocationTargetException, IllegalAccessException {
     ObjectInputFilterApi api = mock(ObjectInputFilterApi.class);
-    GlobalSerialFilterFactory factory = new ReflectiveFacadeGlobalSerialFilterFactory(api);
-    GlobalSerialFilter filter = factory.create("pattern", emptySet());
+    ReflectiveFacadeGlobalSerialFilterFactory factory =
+        new ReflectiveFacadeGlobalSerialFilterFactory();
+    GlobalSerialFilter filter = factory.create(api, "pattern", emptySet());
```

Comment:
This tests an implementation detail, and leaves the actual responsibility (invoked via the other `create()` method) untested.

My sense is that it's just as easy to pass the api through the constructor, and that there's no harm making that constructor visible for even non-test code. Even if you want to restrict that constructor only to tests, invoking the real `create()` here exercises the responsibility more fully (everything except supplying the api used in production).

Reply:
Yeah, after reviewing this test class myself, I'm really not happy with where it's currently at.

Reply:
Fixed up ReflectiveFacadeGlobalSerialFilterFactoryTest